src/plugins/ssh-agent/index.ts

```diff
@@ -59,100 +56,71 @@ const asyncSpawn = async (
     }
   });
 
-/** Spawns a subprocess with the ssh-agent command.
- * Detects the auth socket and agent PID from stdout.
- * Stdout and stderr of the subprocess is printed to stderr in debug mode.
- * The returned promise resolves with an object that contains the auth socket and agent PID,
- * or rejects with the contents of stderr. */
-export const sshAgent = async (cmdArgs: AgentArgs) =>
-  new Promise<SshAgentEnv>((resolve, reject) => {
-    let stderr = "";
-    let stdout = "";
-
-    // There is a debug flag in ssh-agent but it causes the ssh-agent process to NOT fork
-    const child = spawn("ssh-agent");
-
-    child.stdout.on("data", (data) => {
-      const str = data.toString("utf-8");
-      if (cmdArgs.debug) {
-        print2(str);
-      }
-      stdout += str;
-    });
-
-    child.stderr.on("data", (data) => {
-      const str = data.toString("utf-8");
-      if (cmdArgs.debug) {
-        print2(str);
-      }
-      stderr += str;
-    });
-
-    const exitListener = child.on("exit", (code) => {
-      exitListener.unref();
-
-      if (code !== 0) {
-        return reject(stderr);
-      }
-
-      const authSockMatch = stdout.match(AUTH_SOCK_MESSAGE);
-      const agentPidMatch = stdout.match(AGENT_PID_MESSAGE);
+const isSshAgentRunning = async (args: AgentArgs) => {
+  try {
+    if (args.debug) print2("Searching for active ssh-agents");
+    await asyncSpawn(args, `pgrep`, ["-x", "ssh-agent"]);
+    if (args.debug) print2("At least one SSH agent is running");
+    return true;
+  } catch {
+    if (args.debug) print2("No SSH agent is running!");
+    return false;
+  }
+};
 
-      if (!authSockMatch?.[1] || !agentPidMatch?.[1]) {
-        return reject("Failed to parse ssh-agent stdout:\n" + stdout);
-      }
-      resolve({
-        SSH_AUTH_SOCK: authSockMatch[1],
-        SSH_AGENT_PID: agentPidMatch[1],
-      });
-    });
-  });
+const isSshAgentAuthSocketSet = async (args: AgentArgs) => {
```

Comment:
Is the auth socket env var enough? Running ssh-agent also sets `SSH_AGENT_PID`, isn't that needed?

Reply:
Yeah it turns out that we don't need a PID, as long as the auth socket is set everything works fine. 👍